From c013e76b1cf56b89b5faf941c8a5441433f355b8 Mon Sep 17 00:00:00 2001
From: Jonh Wendell <jonh.wendell@redhat.com>
Date: Tue, 8 Feb 2022 16:02:21 -0500
Subject: Restore proxyv2 Dockerfile

For now we are still using upstream proxy, so we must use their
Dockerfile. Once we switch to maistra proxy, we can revert this.
---
 pilot/docker/Dockerfile.proxyv2               | 26 +++++++++++++------
 ...ile.proxyv2 => Dockerfile.proxyv2.maistra} |  0
 2 files changed, 18 insertions(+), 8 deletions(-)
 copy pilot/docker/{Dockerfile.proxyv2 => Dockerfile.proxyv2.maistra} (100%)

diff --git a/pilot/docker/Dockerfile.proxyv2 b/pilot/docker/Dockerfile.proxyv2
index 7680d71b74..02cded141b 100644
--- a/pilot/docker/Dockerfile.proxyv2
+++ b/pilot/docker/Dockerfile.proxyv2
@@ -1,9 +1,22 @@
-FROM centos:8
+# BASE_DISTRIBUTION is used to switch between the old base distribution and distroless base images
+ARG BASE_DISTRIBUTION=debug
 
-# hadolint ignore=DL3039,DL3041
-RUN dnf -y upgrade --refresh && \
-    dnf -y install iptables iproute openssl && \
-    dnf -y clean all
+# Version is the base image version from the TLD Makefile
+ARG BASE_VERSION=latest
+
+# The following section is used as base image if BASE_DISTRIBUTION=debug
+FROM gcr.io/istio-release/base:${BASE_VERSION} as debug
+
+# The following section is used as base image if BASE_DISTRIBUTION=distroless
+# This image is a custom built debian11 distroless image with multiarchitecture support.
+# It is built on the base distroless image, with iptables binary and libraries added
+# The source can be found at https://github.com/istio/distroless/tree/iptables
+# This version is from commit 20b20ec36f4621830ff0bfdec519577074df47ca.
+FROM gcr.io/istio-release/iptables@sha256:bae9287d64be13179b7bc794ec3db26bd5c5fe3fb591c484992366314c9a7d3d as distroless
+
+# This will build the final image based on either debug or distroless from above
+# hadolint ignore=DL3006
+FROM ${BASE_DISTRIBUTION:-debug}
 
 WORKDIR /
 
@@ -15,9 +28,6 @@ ARG SIDECAR=envoy
 COPY envoy_bootstrap.json /var/lib/istio/envoy/envoy_bootstrap_tmpl.json
 COPY gcp_envoy_bootstrap.json /var/lib/istio/envoy/gcp_envoy_bootstrap_tmpl.json
 
-RUN useradd -m --uid 1337 istio-proxy
-RUN chown -R istio-proxy /var/lib/istio
-
 # Install Envoy.
 ARG TARGETARCH
 COPY ${TARGETARCH:-amd64}/${SIDECAR} /usr/local/bin/${SIDECAR}
diff --git a/pilot/docker/Dockerfile.proxyv2 b/pilot/docker/Dockerfile.proxyv2.maistra
similarity index 100%
copy from pilot/docker/Dockerfile.proxyv2
copy to pilot/docker/Dockerfile.proxyv2.maistra
-- 
2.35.3

